chia/full_node/mempool_manager.py

```diff
@@ -80,6 +93,84 @@ def validate_clvm_and_signature(
     return None, bytes(result), new_cache_entries
 
 
+def run_for_cost_and_additions(
+    coin_id: bytes32, puzzle_reveal: SerializedProgram, solution: SerializedProgram, maximum_cost: uint64
+) -> Optional[Tuple[uint64, List[Coin]]]:
+    try:
```

Comment:
if you make `check_item_for_dedup()` signal errors via an exception instead of returning `None`, you can make this function `run_for_cost_and_additions()` also signal errors via an exception instead of returning `None`. i.e. you could remove this `try`-`catch` block